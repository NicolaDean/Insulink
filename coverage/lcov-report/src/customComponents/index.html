
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/customComponents</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> src/customComponents</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">54.78% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>63/115</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">46.96% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>31/66</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">23.25% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>10/43</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">55.26% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>63/114</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line medium'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="PopUp.js"><a href="PopUp.js.html">PopUp.js</a></td>
	<td data-value="48.88" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 48%"></div><div class="cover-empty" style="width: 52%"></div></div>
	</td>
	<td data-value="48.88" class="pct low">48.88%</td>
	<td data-value="45" class="abs low">22/45</td>
	<td data-value="36" class="pct low">36%</td>
	<td data-value="25" class="abs low">9/25</td>
	<td data-value="20" class="pct low">20%</td>
	<td data-value="20" class="abs low">4/20</td>
	<td data-value="48.88" class="pct low">48.88%</td>
	<td data-value="45" class="abs low">22/45</td>
	</tr>

<tr>
	<td class="file medium" data-value="activityChart.js"><a href="activityChart.js.html">activityChart.js</a></td>
	<td data-value="55.55" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 55%"></div><div class="cover-empty" style="width: 45%"></div></div>
	</td>
	<td data-value="55.55" class="pct medium">55.55%</td>
	<td data-value="9" class="abs medium">5/9</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="3" class="abs low">0/3</td>
	<td data-value="55.55" class="pct medium">55.55%</td>
	<td data-value="9" class="abs medium">5/9</td>
	</tr>

<tr>
	<td class="file medium" data-value="calendarPicker.js"><a href="calendarPicker.js.html">calendarPicker.js</a></td>
	<td data-value="61.9" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 61%"></div><div class="cover-empty" style="width: 39%"></div></div>
	</td>
	<td data-value="61.9" class="pct medium">61.9%</td>
	<td data-value="21" class="abs medium">13/21</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="8" class="abs medium">4/8</td>
	<td data-value="65" class="pct medium">65%</td>
	<td data-value="20" class="abs medium">13/20</td>
	</tr>

<tr>
	<td class="file high" data-value="customButton.js"><a href="customButton.js.html">customButton.js</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="88.88" class="pct high">88.88%</td>
	<td data-value="9" class="abs high">8/9</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	</tr>

<tr>
	<td class="file high" data-value="customImageButton.js"><a href="customImageButton.js.html">customImageButton.js</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="91.66" class="pct high">91.66%</td>
	<td data-value="12" class="abs high">11/12</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	</tr>

<tr>
	<td class="file medium" data-value="glycemiaChart.js"><a href="glycemiaChart.js.html">glycemiaChart.js</a></td>
	<td data-value="50" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 50%"></div><div class="cover-empty" style="width: 50%"></div></div>
	</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="10" class="abs medium">5/10</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="3" class="abs low">0/3</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="10" class="abs medium">5/10</td>
	</tr>

<tr>
	<td class="file low" data-value="macroChart.js"><a href="macroChart.js.html">macroChart.js</a></td>
	<td data-value="42.85" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 42%"></div><div class="cover-empty" style="width: 58%"></div></div>
	</td>
	<td data-value="42.85" class="pct low">42.85%</td>
	<td data-value="14" class="abs low">6/14</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="10" class="abs low">0/10</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="42.85" class="pct low">42.85%</td>
	<td data-value="14" class="abs low">6/14</td>
	</tr>

<tr>
	<td class="file medium" data-value="sportChart.js"><a href="sportChart.js.html">sportChart.js</a></td>
	<td data-value="60" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 60%"></div><div class="cover-empty" style="width: 40%"></div></div>
	</td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="10" class="abs medium">6/10</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="3" class="abs low">0/3</td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="10" class="abs medium">6/10</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at Thu May 26 2022 11:26:03 GMT+0200 (Central European Summer Time)
            </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    