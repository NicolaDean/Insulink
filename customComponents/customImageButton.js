
import React from 'react';
import { Text, StyleSheet,TouchableOpacity, View, Image } from 'react-native';

const images = {
    plus: {
      uri: require('../assets/plus.png')
    }
  }

export const CustomButton = (
    {   onPress,
        title="Button",
         image="plus",
        style,
        size=""
    }) =>{

        return(
            <View style={style}>
                    <TouchableOpacity onPress={onPress} style={{}}>
                        <Image source={images[image].uri}  style={{width: 40, height: 40}}  />
                    </TouchableOpacity>
            </View>
           
        );
}

const styles =  StyleSheet.create({
    appButtonText:{
        fontSize: 18,
        color: "#fff",
        fontWeight: "bold",
        alignSelf: "center",
        textTransform: "uppercase"
      }
});

export default CustomButton;