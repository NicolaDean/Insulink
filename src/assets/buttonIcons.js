
export const buttonIcons = {
    plus:{uri: require('./plus.png')},
    camera:{uri: require('./camera.png')},
    search: {uri: require('./search.png')},
    delete: {uri: require('./delete.png')},
    close: {uri: require('./close.png')},
    running: {uri: require('./running.png')},
    soccer: {uri: require('./soccer.png')},
    tennis: {uri: require('./tennis.png')},
    fitness: {uri: require('./fitness.png')},
    basket: {uri: require('./basket.png')},
    swimming: {uri: require('./swimming.png')},
    more: {uri: require('./more.png')},
    less: {uri: require('./less.png')},
    ok: {uri: require('./ok.png')},
    settings: {uri: require('./settings.png')},
    return:{uri:require('./return.png')},
    dose: {uri: require('./dose.png')},
    glucose: {uri: require('./glucose.png')},
    statistics:{uri: require('./statistics.png')},
    calendar:{uri: require('./calendar.png')},
    height:{uri: require('./height.png')},
    weigth:{uri: require('./weight.png')},
    male:{uri: require('./male.png')},
    female:{uri: require('./female.png')},
    nonbinary:{uri: require('./nonbinary.png')},
}

export const buttonIconsNames = {
    plus:'plus',
    camera:'camera',
    search: 'search',
    delete: 'delete',
    close: 'close',
    running: 'running',
    soccer: 'soccer',
    tennis: 'tennis',
    fitness: 'fitness',
    basket: 'basket',
    swimming: 'swimming',
    more: 'more',
    less:'less',
    ok:'ok',
    settings:'settings',
    return:'return',
    does: 'dose',
    glucose: 'glucose',
    statistics: 'statistics',
    calendar: 'calendar',
    height:'height',
    weigth:'weight',
    male:'male',
    female:'female',
    nonbinary:'nonbinary'
}

