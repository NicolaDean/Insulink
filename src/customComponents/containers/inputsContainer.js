import React, { useState } from 'react';
import { Text, StyleSheet,TouchableOpacity, TextInput,View} from 'react-native';
import { colors } from '../../constants/appAspect';

export const InputContainer = ({children,name,style,childrenStyle}) =>{

    return(
        <View style={[styles.container,style]}>
            <Text style={styles.name}>{name}</Text>
            <View style={[styles.children,childrenStyle]}>
                {children}
            </View>
            
        </View>
    );
}

const styles = StyleSheet.create({
    container:{
        marginLeft:10,
        marginTop:10,
        flexDirection:'column',
        justifyContent:'center',
    },
    name:{
        fontSize:18,
        fontWeight:'600',
        /*alignSelf:'center'*/
        marginLeft:10,
        color:colors.primary,
    },
    children:{

    }
})