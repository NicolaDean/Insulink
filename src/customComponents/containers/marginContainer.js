import React, { useState } from 'react';
import { Text, StyleSheet,TouchableOpacity, TextInput,View} from 'react-native';

export const MarginContainer = ({children,style={},width}) =>{

    return(
        <View style={[styles.container,style,{width:width}]}>
            {children}
        </View>
    );
}

const styles = StyleSheet.create({
    container:{
        width:'90%',
        marginLeft:'5%',
        marginTop:10,
    }
})