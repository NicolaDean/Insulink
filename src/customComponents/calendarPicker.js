import React, { useState } from 'react';
import { Text, Image,StyleSheet,TouchableOpacity, View,LayoutAnimation} from 'react-native';
import { colors } from '../constants/appAspect';
import CalendarPicker from 'react-native-calendar-picker';
import { buttonIcons } from '../assets/buttonIcons';
import { FirebaseQuery } from '../utils/firebaseQuery';


export default Calendar = ({onChange = (date)=>{}}) => {

    const [currentDate,setDate] = useState(FirebaseQuery.printFormattedDate(new Date()));
    const [isexpanded,setExpanded] = useState(false);

    const expandCalendar = () =>{
        LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut); 
        setExpanded(expanded => !expanded); 
    }
    
    const expanded = () =>{
        return(
            <View>
                <CalendarPicker
                    onDateChange={onDateChange}
                />
            </View>
        );
    }

    const notExpanded = () =>{
        return(
            <View>
            </View>
        );
    }

    const onDateChange = (date) =>{
        //TODO ADD A REDUX ACTION TO RETRIVE DATA FROM LOCALSTORAGE/FIREBASE AND SHOW THEM
        const parseDate = new Date(date);
        //let currDate = FirebaseQuery.glicemyDateFormatter(parseDate);
        let string_format = FirebaseQuery.printFormattedDate(parseDate);
        let storage_format= FirebaseQuery.glicemyDateFormatter(parseDate);

        onChange(storage_format);
        setDate(string_format);
    }

    return (
        <View style={{justifyContent:'center',alignContent:'center',alignSelf:'center',flexDirection:'column'}}>
            <TouchableOpacity onPress={expandCalendar}>
                <View style={{justifyContent:'space-evenly',alignContent:'stretch',alignSelf:'center',flexDirection:'row',marginVertical:10}}>
                <Text style={{alignSelf:'center',marginHorizontal:10}} >{currentDate.toString()}</Text>
                <Image source={buttonIcons['calendar'].uri} style={styles.icon} /> 
                </View>
                {isexpanded ? expanded():notExpanded()}
            </TouchableOpacity>
        </View>
    );
}

const styles =  StyleSheet.create({
    icon:{
         width: 32,
         height: 32,
         position: 'relative',
         top:0,
         bottom:10,
        }
});