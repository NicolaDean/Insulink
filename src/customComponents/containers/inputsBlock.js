import React, { useState } from 'react';
import { Text, StyleSheet,TouchableOpacity, TextInput,View} from 'react-native';
import { MarginContainer } from './marginContainer';

export const InputBlock = ({children,name}) =>{

    return(
        <View style={[styles.container]}>
            <MarginContainer>
                <Text style={styles.title}>{name}</Text>
                <View style={[styles.inputs]}>
                    {children}
                </View>
            </MarginContainer>     
        </View>
        
    );
}

const styles = StyleSheet.create({
    container:{
        marginTop:10,
        backgroundColor:'white'
    },
    title:{
        fontSize:20,
    },
    inputs:{
        width:'100%',
        flexDirection:'row',
    }
})