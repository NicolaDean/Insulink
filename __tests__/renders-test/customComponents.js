import React, { useState } from 'react';
import renderer from 'react-test-renderer';
import { buttonIcons, buttonIconsNames } from '../../src/assets/buttonIcons';
import { Calendar } from '../../src/customComponents/calendarPicker';
import { InputBlock } from '../../src/customComponents/containers/inputsBlock';
import { InputContainer } from '../../src/customComponents/containers/inputsContainer';
import { MarginContainer } from '../../src/customComponents/containers/marginContainer';
import { WaitLoading } from '../../src/customComponents/containers/waitLoading';
import CustomButton from '../../src/customComponents/customButton'
import  { CustomImageButton } from '../../src/customComponents/customImageButton'

const fir = ()=>{
    return {
      doc:null,
      collection: (a) =>{return "ciao"},
    };
}


jest.mock("@react-native-firebase/firestore", () => fir);
jest.mock('react-native-calendar-picker', () => 'CalendarPicker');

/**
 * Ensure that all custom components (eg. custom buttons,calendar,style containers) remain equal among code edits
 */
describe("Render All Custom Components: ", () => {

  beforeEach(() => {
    //Executed before each test
  });
  
  test('Render Calendar', () => {

    const tree = renderer.create(<Calendar openC={true}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Render Custom Button', () => {

    const tree = renderer.create(
    <CustomButton title ="bottone" 
                  onPress = {()=>{console.log("test")}}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Render Custom Image Button', () => {
    const tree = renderer.create(
          <CustomImageButton image={buttonIconsNames.plus} 
                             style={{}} 
                             iconStyle={{width:50,height:50}} 
                             onPress = {()=>{console.log("test")}}/>
          ).toJSON();
    expect(tree).toMatchSnapshot();
  });
})


describe("Render All Custom Container: ", () => {

  beforeEach(() => {
    //Executed before each test
  });

  test('Render InputsBlock', () => {

    const tree = renderer.create(
    <InputBlock>
        <CustomButton title ="bottone" onPress = {()=>{console.log("test")}}/>
    </InputBlock>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Render InputsContainer', () => {

    const tree = renderer.create(
    <InputContainer>
        <InputBlock>
            <CustomButton title ="first" onPress = {()=>{console.log("test")}}/>
        </InputBlock>
        <InputBlock>
          <CustomButton title ="second" onPress = {()=>{console.log("test")}}/>
       </InputBlock>
    </InputContainer>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });


  test('Render MarginContainer', () => {

    const tree = renderer.create(

      <MarginContainer>
          <CustomButton title ="second" onPress = {()=>{console.log("test")}}/>
      </MarginContainer>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Render WaitLoading', () => {

    let [loading,setLoading] = [false,()=>{}];
    const tree = renderer.create(

      <WaitLoading loadingState={[loading,setLoading]}>
          <CustomButton title ="Button Appeared, No Loading" onPress = {()=>{console.log("test")}}/>
      </WaitLoading>
    ).toJSON();
    expect(tree).toMatchSnapshot();

    [loading,setLoading] = [true,()=>{}];

    const tree2 = renderer.create(

      <WaitLoading loadingState={[loading,setLoading]}>
          <CustomButton title ="Should not appear" onPress = {()=>{console.log("test")}}/>
      </WaitLoading>
    ).toJSON();
    expect(tree2).toMatchSnapshot();
  });
  

});