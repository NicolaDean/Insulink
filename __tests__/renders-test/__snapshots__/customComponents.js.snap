// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render All Custom Components:  Render Custom Button 1`] = `
<View>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "rgb(112,202,230)",
        "borderRadius": 10,
        "elevation": 0,
        "margin": 20,
        "opacity": 1,
        "paddingHorizontal": 12,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#fff",
            "fontSize": 18,
            "fontWeight": "bold",
            "textTransform": "uppercase",
          },
          null,
        ]
      }
    >
      bottone
    </Text>
  </View>
</View>
`;

exports[`Render All Custom Components:  Render Custom Image Button 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="ImageButton"
>
  <View
    style={null}
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/plus.png",
        }
      }
      style={
        Object {
          "height": 50,
          "width": 50,
        }
      }
    />
  </View>
</View>
`;

exports[`Render All Custom Container:  Render InputsBlock 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#285AA4",
        "borderRadius": 8,
        "elevation": 50,
        "marginTop": 10,
        "paddingHorizontal": 12,
        "paddingVertical": 25,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginLeft": "5%",
          "marginTop": 10,
          "width": "90%",
        },
        Object {},
        Object {
          "width": "90%",
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "flexDirection": "column",
            "width": "100%",
          },
        ]
      }
    >
      <View>
        <View
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "rgb(112,202,230)",
              "borderRadius": 10,
              "elevation": 0,
              "margin": 20,
              "opacity": 1,
              "paddingHorizontal": 12,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "color": "#fff",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "textTransform": "uppercase",
                },
                null,
              ]
            }
          >
            bottone
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Render All Custom Container:  Render InputsContainer 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "column",
        "justifyContent": "center",
        "marginLeft": 10,
        "marginTop": 5,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "color": "#fff",
        "fontSize": 18,
        "fontWeight": "600",
        "marginLeft": 10,
      }
    }
  />
  <View
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#285AA4",
            "borderRadius": 8,
            "elevation": 50,
            "marginTop": 10,
            "paddingHorizontal": 12,
            "paddingVertical": 25,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "marginLeft": "5%",
              "marginTop": 10,
              "width": "90%",
            },
            Object {},
            Object {
              "width": "90%",
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 20,
              "fontWeight": "bold",
            }
          }
        />
        <View
          style={
            Array [
              Object {
                "flexDirection": "column",
                "width": "100%",
              },
            ]
          }
        >
          <View>
            <View
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "backgroundColor": "rgb(112,202,230)",
                  "borderRadius": 10,
                  "elevation": 0,
                  "margin": 20,
                  "opacity": 1,
                  "paddingHorizontal": 12,
                  "paddingVertical": 10,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "alignSelf": "center",
                      "color": "#fff",
                      "fontSize": 18,
                      "fontWeight": "bold",
                      "textTransform": "uppercase",
                    },
                    null,
                  ]
                }
              >
                first
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#285AA4",
            "borderRadius": 8,
            "elevation": 50,
            "marginTop": 10,
            "paddingHorizontal": 12,
            "paddingVertical": 25,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "marginLeft": "5%",
              "marginTop": 10,
              "width": "90%",
            },
            Object {},
            Object {
              "width": "90%",
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 20,
              "fontWeight": "bold",
            }
          }
        />
        <View
          style={
            Array [
              Object {
                "flexDirection": "column",
                "width": "100%",
              },
            ]
          }
        >
          <View>
            <View
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "backgroundColor": "rgb(112,202,230)",
                  "borderRadius": 10,
                  "elevation": 0,
                  "margin": 20,
                  "opacity": 1,
                  "paddingHorizontal": 12,
                  "paddingVertical": 10,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "alignSelf": "center",
                      "color": "#fff",
                      "fontSize": 18,
                      "fontWeight": "bold",
                      "textTransform": "uppercase",
                    },
                    null,
                  ]
                }
              >
                second
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Render All Custom Container:  Render MarginContainer 1`] = `
<View
  style={
    Array [
      Object {
        "marginLeft": "5%",
        "marginTop": 10,
        "width": "90%",
      },
      Object {},
      Object {
        "width": "90%",
      },
    ]
  }
>
  <View>
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgb(112,202,230)",
          "borderRadius": 10,
          "elevation": 0,
          "margin": 20,
          "opacity": 1,
          "paddingHorizontal": 12,
          "paddingVertical": 10,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "alignSelf": "center",
              "color": "#fff",
              "fontSize": 18,
              "fontWeight": "bold",
              "textTransform": "uppercase",
            },
            null,
          ]
        }
      >
        second
      </Text>
    </View>
  </View>
</View>
`;

exports[`Render All Custom Container:  Render WaitLoading 1`] = `
<View>
  <View>
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgb(112,202,230)",
          "borderRadius": 10,
          "elevation": 0,
          "margin": 20,
          "opacity": 1,
          "paddingHorizontal": 12,
          "paddingVertical": 10,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "alignSelf": "center",
              "color": "#fff",
              "fontSize": 18,
              "fontWeight": "bold",
              "textTransform": "uppercase",
            },
            null,
          ]
        }
      >
        Button Appeared, No Loading
      </Text>
    </View>
  </View>
</View>
`;

exports[`Render All Custom Container:  Render WaitLoading 2`] = `
<View>
  <ActivityIndicator />
</View>
`;
